### ISSUE TITLE
google_compute_vpn_gateway does not work if region not specified

### ISSUE THREAD

**Issue by clstokes** [hashicorp/terraform#5027]
When trying to create a `google_compute_vpn_gateway` resource, if the `region` is not specified, Terraform will create the VPN gateway, but is then unable to _read_ it ([a 404 is returned](https://github.com/hashicorp/terraform/blob/1188f1ba5a6ec391e6158b51021a2dd96b46bc34/builtin/providers/google/resource_compute_vpn_gateway.go#L90)) and no error is reported. At this point, the gateway has been created, but the gateway's details are not persisted in Terraform's state. Subsequent applies and destroys will fail.

If I specify the region (commented below), Terraform works fine. It seems that [resourceComputeVpnGatewayRead()](https://github.com/hashicorp/terraform/blob/1188f1ba5a6ec391e6158b51021a2dd96b46bc34/builtin/providers/google/resource_compute_vpn_gateway.go#L86) should be reading the region from [getOptionalRegion()](https://github.com/hashicorp/terraform/blob/1188f1ba5a6ec391e6158b51021a2dd96b46bc34/builtin/providers/google/resource_compute_vpn_gateway.go#L55) as other Google resources do. In fact, changing this locally fixes this, but I'm unsure if there are other implications of this change.

###### google_compute_vpn_gateway
```
resource "google_compute_network" "main" {
  name = "tf-test"
  ipv4_range = "172.17.0.0/16"
}

resource "google_compute_address" "vpn" {
  name = "tf-test"
}

resource "google_compute_vpn_gateway" "vpn" {
  name = "tf-test"
  network = "${google_compute_network.main.self_link}"
  #region = "us-central1"
}
```

###### google_compute_vpn_tunnel
This occurs for `google_compute_vpn_tunnel` too. Full config to reproduce is at https://gist.github.com/clstokes/7b89b5542c02deedddc6.

###### Docs issue (?)
The [compute_vpn_gateway docs](https://github.com/hashicorp/terraform/blob/1188f1ba5a6ec391e6158b51021a2dd96b46bc34/website/source/docs/providers/google/r/compute_vpn_gateway.html.markdown) say _If not specified, the project region will be used._ where other resources say _If it is not provided, the provider region is used._ I'm not sure if this is a typo or there actually is a distinction between **project** and **provider** region.

**Comment by clstokes** [comment]
Edited to include details about `google_compute_vpn_tunnel` too.

**Comment by lwander** [comment]
Good catch! Submitted a fix #5125

**Comment by jen20** [comment]
Fixed by #5125 - nice catch @clstokes!
